clpy/random/generator.py

```diff
@@ -172,15 +172,18 @@ def random_sample(self, size=None, dtype=float):
         out = clpy.empty(size, dtype=dtype)
 
         if (not isinstance(self.seed_array, clpy.ndarray)
-                or self.seed_array.size != numpy.prod(size)):
+                or self.seed_array.size < numpy.prod(size)):
             self.seed_array = clpy.empty(size, "uint")
             tmp_seed_array = clpy.empty(size, "uint")
             tmp_seed_array.fill(self.seed_value)
             RandomState._init_kernel(tmp_seed_array, self.seed_array)
             # not to use similar number for the first generation
             RandomState._lcg_kernel(self.seed_array, out)
-
-        RandomState._lcg_kernel(self.seed_array, out)
+            RandomState._lcg_kernel(self.seed_array, out)
+        else:
+            tmp = self.seed_array.reshape(self.seed_array.size)[
+                0:numpy.prod(size)].reshape(size)
```

Comment:
`numpy.prod(size)` was calculared also on L175. How about name it as a variable?

Reply:
`numpy.prod(size)` is assigned to `array_size`. 

I also commented why this is required since it is not clear from the name of 'size'.